src/library_signals.js

```diff
@@ -5,91 +5,42 @@
  */
 
 // 'use strict'
-var funs = {
+var LibrarySignals = {
   _sigalrm_handler: 0,
 
-  signal__deps: ['_sigalrm_handler'],
-  signal: function(sig, func) {
+  __sigaction__deps: ['_sigalrm_handler'],
+  __sigaction: function(sig, act, oldact) {
+    //int sigaction(int signum, const struct sigaction *act, struct sigaction *oldact);
     if (sig == {{{ cDefine('SIGALRM') }}}) {
-      __sigalrm_handler = func;
-    } else {
-#if ASSERTIONS
-      err('Calling stub instead of signal()');
-#endif
+      __sigalrm_handler = {{{ makeGetValue('act', '0', 'i32') }}};
+      return 0;
     }
-    return 0;
-  },
-  bsd_signal__sig: 'iii',
-  bsd_signal: 'signal',
```

Comment:
does musl provide `bsd_signal`? it seems like no as I can't find it in the headers.

Reply:
Yup, its provided as an alias of signal:

```
$ git grep bsd_signal system/lib/libc
system/lib/libc/musl/include/signal.h:void (*bsd_signal(int, void (*)(int)))(int);
system/lib/libc/musl/src/signal/signal.c:weak_alias(signal, bsd_signal);
```